hax/hax/motr/__init__.py

```diff
@@ -281,13 +281,35 @@ def ha_obj_state(st):
             if (st.fid.container == ObjT.PROCESS.value
                     and st.status == ServiceHealth.STOPPED):
                 notify_devices = False
+
+            if st.fid.container == ObjT.PROCESS.value:
+                self.consul_util.set_process_state(st.fid, st.status)
+
             notes += self._generate_sub_services(note, self.consul_util,
                                                  notify_devices)
             # For process failure, we report failure for the corresponding
-            # node (enclosure) and CVGs.
+            # node (enclosure) and CVGs if all Io services are failed.
             if (st.fid.container == ObjT.PROCESS.value
                     and st.status in (ServiceHealth.FAILED, ServiceHealth.OK)):
-                notes += self.notify_node_status(note)
+                # Check if we need to mark node as failed,
+                # otherwise just mark controller as failed/OK
+                # If we receive process failure then we will check if all IO
+                # services are failed, if True then we ill mark node as failed
+                # If we receive process 'OK' then wwe will check if node is
+                # not in failed state then we ill mark node as OK
+                # If both the above conditions are not true then we will just
+                # mark controller status
+                if (st.status == ServiceHealth.FAILED
+                        and self.check_for_node_failure(note) is True):
```

Comment:
```suggestion
                        and self.check_for_node_failure(note)):
```
Which means that `check_for_node_failure` is not the best name. It seems like the functionality behind corresponds to "Is node failed?" predicate.

Reply:
Done